components/server/src/user/newsletter-subscription-controller.ts

```diff
@@ -85,6 +104,56 @@ export class NewsletterSubscriptionController {
             }
         })
 
+        router.post("/subscribe", async (req: express.Request, res: express.Response) => {
+            try {
+                const data = JSON.parse(req.body);
+                if (!SubscriptionData.is(data)) {
+                    res.status(400).send('Invalid message body');
+                    log.error("Invalid subscribe request", data)
+                    return;
+                }
+                if (!acceptedNewsletterTypes.includes(data.newsletterType)) {
+                    log.error("Invalid newsletter in subscribe request", data)
+                    res.status(400).send('Invalid newsletter type');
+                    return;
+                }
+                const user = (await this.userDb.findUsersByEmail(data.email))[0];
```

Comment:
1. I'm concerned about adding an open endpoint to patch any user's settings like this. Same applies to connecting to 3rd party services. Obviously, there are very simple ways how to abuse this and even disrupt our services. We should consider a verification protocol (or other suitable mechanisms) for protection.

2. Email addresses are not suitable to identify users. `findUsersByEmail` should not be used to select a user in general. We should rename it to discourage usage of it. Currently, it's used to verify email addresses already used in existing accounts.

Reply:
I understand your concerns. What do you propose? We need a way for un-registered users to un-/subscribe to the newsletter.
Should we pull newsletter subscription out of the app (and the db) entirely and handle it in segment/customer.io only?

Reply:
the case of unregistered user in particular isn't clear to me: should we really introducing an open relay for a connected service? also, doesn't subscribing to newsletter require us to send verification emails? 

> Should we pull newsletter subscription out of the app (and the db) entirely and handle it in segment/customer.io only?

If we want to enable newsletter signups for unregistered users, it really looks like a separate service which we then would need to talk to from the web app to adjust settings.